controllers/osccluster_routetable_controller_unit_test.go

```diff
@@ -0,0 +1,3292 @@
+							Routes: []infrastructurev1beta1.OscRoute{
+								{
+									Name:        "test-route",
+									TargetName:  "test-internetservice",
+									TargetType:  "gateway",
+									Destination: "0.0.0.0/0",
+								},
+								{
+									Name:        "test-route",
+									TargetName:  "test-internetservice",
+									TargetType:  "gateway",
+									Destination: "0.0.0.0/0",
+								},
+							},
+						},
+					},
+				},
+			},
+			expCheckRouteOscDuplicateNameErr: fmt.Errorf("test-route already exist"),
+		},
+		{
+			name: "check route duplicate  nat service name",
+			spec: infrastructurev1beta1.OscClusterSpec{
+				Network: infrastructurev1beta1.OscNetwork{
+					Net: infrastructurev1beta1.OscNet{
+						Name:    "test-net",
+						IpRange: "10.0.0.0/16",
+					},
+					Subnets: []*infrastructurev1beta1.OscSubnet{
+						{
+							Name:          "test-subnet",
+							IpSubnetRange: "10.0.0.0/24",
+						},
+					},
+					InternetService: infrastructurev1beta1.OscInternetService{
+						Name: "test-internetservice",
+					},
+					RouteTables: []*infrastructurev1beta1.OscRouteTable{
+						{
+							Name:       "test-routetable",
+							SubnetName: "test-subnet",
+							Routes: []infrastructurev1beta1.OscRoute{
+								{
+									Name:        "test-route",
+									TargetName:  "test-natservice",
+									TargetType:  "nat",
+									Destination: "0.0.0.0/0",
+								},
+								{
+									Name:        "test-route",
+									TargetName:  "test-natservice",
+									TargetType:  "nat",
+									Destination: "0.0.0.0/0",
+								},
+							},
+						},
+					},
+				},
+			},
+			expCheckRouteOscDuplicateNameErr: fmt.Errorf("test-route already exist"),
+		},
+		{
+			name: "check no routetable",
+			spec: infrastructurev1beta1.OscClusterSpec{
+				Network: infrastructurev1beta1.OscNetwork{
+					Net: infrastructurev1beta1.OscNet{
+						Name:    "test-net",
+						IpRange: "10.0.0.0/16",
+					},
+					Subnets: []*infrastructurev1beta1.OscSubnet{
+						{
+							Name:          "test-subnet",
+							IpSubnetRange: "10.0.0.0/24",
+						},
+					},
+					InternetService: infrastructurev1beta1.OscInternetService{
+						Name: "test-internetservice",
+					},
+					RouteTables: []*infrastructurev1beta1.OscRouteTable{
+						{},
+					},
+				},
+			},
+			expCheckRouteOscDuplicateNameErr: nil,
+		},
+	}
+	for _, rtc := range routeTestCases {
+		t.Run(rtc.name, func(t *testing.T) {
+			t.Logf("Validate to %s", rtc.name)
+			log := klogr.New()
+			oscCluster := infrastructurev1beta1.OscCluster{
+				Spec: rtc.spec,
+				ObjectMeta: metav1.ObjectMeta{
+					UID: "uid",
+				},
+			}
+			clusterScope := &scope.ClusterScope{
+				Logger: log,
+				Cluster: &clusterv1.Cluster{
+					ObjectMeta: metav1.ObjectMeta{
+						UID: "uid",
+					},
+				},
+				OscCluster: &oscCluster,
+			}
+			duplicateResourceRouteNameErr := checkRouteOscDuplicateName(clusterScope)
+			if duplicateResourceRouteNameErr != nil {
```

Comment:
If the function does not return an error and if we expect an error, this will not be detected

Reply:
ok done

Reply:
I use assert instead (https://pkg.go.dev/github.com/stretchr/testify/assert)